app/src/main/java/combi/example/andre/finalproject/DataParser.java

```diff
@@ -0,0 +1,82 @@
+package combi.example.andre.finalproject;
+
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class DataParser {
+    public List<HashMap<String, String>> parse(String jsonData){
+        JSONArray jsonArray = null;
+        JSONObject jsonObject;
+
+        try{
+            Log.d("places", "parse");
+            jsonObject = new JSONObject((String) jsonData);
+            jsonArray = jsonObject.getJSONArray("results");
+        }
+        catch(JSONException e){
+            Log.d("Places", "parse error");
+            e.printStackTrace();
+        }
+        return getPlaces(jsonArray);
+    }
+
+    public List<HashMap<String, String>> getPlaces(JSONArray jsonArray){
+        int placesCount = jsonArray.length();
+        List<HashMap<String, String>> placesList = new ArrayList<>();
+        HashMap<String, String> placeMap = null;
+        Log.d("places", "getPlaces");
+
+        for(int i = 0; i < placesCount; i++){
+            try{
+                placeMap = getPlace((JSONObject) jsonArray.get(i));
+                placesList.add(placeMap);
+                Log.d("Places", "Adding Places");
+            }
+            catch (JSONException e){
+                Log.d("Places", "Error adding places");
+                e.printStackTrace();
+            }
+        }
+        return placesList;
+    }
+
+    private HashMap<String, String> getPlace(JSONObject googlePlaceJson){

```

Comment:
Instead of using HashMap, you might want to consider using model instead, since it is basically googlePlaceMap is an object that has multiple attributes.